src/cbmc/bmc.h

```diff
@@ -28,24 +31,57 @@ Author: Daniel Kroening, kroening@kroening.com
 #include <goto-programs/goto_trace.h>
 
 #include <goto-symex/symex_target_equation.h>
+#include <goto-programs/goto_model.h>
 #include <goto-programs/safety_checker.h>
 #include <goto-symex/memory_model.h>
 
 #include "symex_bmc.h"
 
+class cbmc_solverst;
+
+/// \brief Bounded model checking or path exploration for goto-programs
+///
+/// Higher-level architectural information on symbolic execution is
+/// documented in the \ref symex-overview
+/// "Symbolic execution module page".
 class bmct:public safety_checkert
 {
 public:
+  /// \brief Constructor for path exploration with freshly-initialized state
+  ///
+  /// This constructor should be used for symbolically executing a program
+  /// from the entry point with fresh state. There are two main behaviours
+  /// that can happen when constructing an instance of this class:
+  ///
+  /// - If the `--paths` flag in the `options` argument to this
+  ///   constructor is `false` (unset), an instance of this class will
+  ///   symbolically execute the entire program, performing path merging
+  ///   to build a formula corresponding to all executions of the program
+  ///   up to the unwinding limit. In this case, the `branch_worklist`
+  ///   member shall not be touched; this is enforced by the assertion in
+  ///   this class' implementation of bmct::perform_symbolic_execution().
+  ///
+  /// - If the `--paths` flag is `true`, this `bmct` object will explore a
+  ///   single path through the codebase without doing any path merging.
+  ///   If some paths were not taken, the state at those branch points
+  ///   will be appended to `branch_worklist`. After the single path that
+  ///   this `bmct` object executed has been model-checked, you can resume
+  ///   exploring further paths by popping an element from
+  ///   `branch_worklist` and using it to construct a path_explorert
+  ///   object.
   bmct(
     const optionst &_options,
-    const symbol_tablet &_symbol_table,
+    const symbol_tablet &outer_symbol_table,
     message_handlert &_message_handler,
-    prop_convt &_prop_conv)
+    prop_convt &_prop_conv,
+    goto_symext::branch_worklistt &_branch_worklist)
```

Comment:
For more backwards compatibility, would it have been possible to provide a constructor that initialises branch_worklist to a default value here?

Reply:
I'm not sure default values and references would be compatible?